packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/remote_constraint_provider.dart

```diff
@@ -33,9 +42,99 @@ class DefaultRemoteLoggingConstraintProvider
   final DefaultRemoteConfiguration _config;
   final AWSCredentialsProvider _credentialsProvider;
 
+  LoggingConstraint? _loggingConstraint;
+  DateTime? _lastUpdated;
+
+  /// The signer to sign the request.
+  static const signer = AWSSigV4Signer();
+
+  /// The scope to sign the request.
+  final scope = AWSCredentialScope(region: 'us-west-2', service: AWSService.s3);
+
+  static final _logger = AmplifyLogger('default-remote-config');
+
+  Timer? _timer;
+
+  Future<void> _saveConstraintLocally(LoggingConstraint constraint) async {
+    await storage.saveConstraintLocally(constraint.toJson());
+  }
+
+  Future<void> _fetchAndCacheConstraintFromEndpoint() async {
+    try {
+      final constraint = await _fetchConstraintFromEndpoint();
+      if (constraint != null) {
+        _loggingConstraint = constraint;
+        _lastUpdated = DateTime.now();
+        await _saveConstraintLocally(constraint);
+      }
+      // ignore: avoid_catches_without_on_clauses
+    } catch (error) {
+      _logger.debug('Error fetching constraints: $error');
+    }
+  }
+
+  Future<LoggingConstraint?> _fetchConstraintFromEndpoint() async {
+    final uri = Uri.parse(_config.endpoint);
+
+    final request = AWSHttpRequest(method: AWSHttpMethod.get, uri: uri);
+
+    final signedRequest = await signer.sign(
+      request,
+      credentialScope: scope,
+    );
+
+    final headers = signedRequest.headers.map<String, String>(
+      MapEntry<String, String>.new,
+    );
+
+    final response = await http.get(uri, headers: headers);
+
+    // final response = await http.get(Uri.parse(_config.endpoint));
+    if (response.statusCode == 200) {
+      final fetchedConstraint = LoggingConstraint.fromJson(
+        jsonDecode(response.body) as Map<String, dynamic>,
+      );
+      return fetchedConstraint;
+    }
+    throw Exception(
+      'Failed to fetch logging constraint from ${_config.endpoint}',
+    );
+  }
+
   @override
-  // TODO(nikahsn): add implementation.
-  LoggingConstraint get loggingConstraint => throw UnimplementedError();
+  LoggingConstraint? get loggingConstraint {
+    if (_loggingConstraint != null &&
+        _lastUpdated != null &&
+        DateTime.now().difference(_lastUpdated!).inSeconds <=
+            _config.refreshIntervalInSeconds.inSeconds) {
```

Comment:
not sure if this check is needed. cache is always updated after fetching the constratin from endpoint